src/Propel/Generator/Manager/MigrationManager.php

```diff
@@ -150,34 +160,27 @@ public function getAllDatabaseVersions(): array
             throw new Exception('You must define database connection settings in a buildtime-conf.xml file to use migrations');
         }
 
-        /** @var array<int> $migrationTimestamps */
         $migrationTimestamps = [];
         foreach ($connections as $name => $params) {
-            $conn = $this->getAdapterConnection($name);
-            $platform = $this->getGeneratorConfig()->getConfiguredPlatform($conn);
-            if (!$platform->supportsMigrations()) {
-                continue;
-            }
-
-            $sql = sprintf('SELECT version FROM %s', $this->getMigrationTable());
-
             try {
-                $stmt = $conn->prepare($sql);
-                $stmt->execute();
-
-                while ($migrationTimestamp = $stmt->fetchColumn()) {
-                    /** @phpstan-var int $migrationTimestamp */
-                    $migrationTimestamps[] = $migrationTimestamp;
-                }
+                $migrationTimestamps += $this->getMigrationData($name);
             } catch (PDOException $e) {
                 $this->createMigrationTable($name);
                 $migrationTimestamps = [];
             }
         }
 
-        sort($migrationTimestamps);
+        usort($migrationTimestamps, function (array $a, array $b) {
+            if ($a[static::COL_EXECUTION_DATETIME] === $b[static::COL_EXECUTION_DATETIME]) {
```

Comment:
According to the investigation that was made by @valerio8787, Propel migrate down command may roll back migration in the wrong order. E.g. we have two releases of the application, which contain the following migrations:
```
release 1 (migration applied 01.05.2022):
migration_001.php
migration_008.php
```
```
release 2 (migration applied 01.06.2022):
migration_005.php (migration created during development earlier than migration_008.php from release 1 and has an earlier version)
```

After the deployment of all versions, Propel migration table will have the following state:
```
version
--
001
008
005
```

Now if we run `propel migrate:down` command (that rollback only one migration) it will roll back migration with version 008 first, as Propel orders executed migrations by version and then rolls back a migration with max version value.